src/main/java/seedu/address/model/deck/Deck.java

```diff
@@ -0,0 +1,84 @@
+package seedu.address.model.deck;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+
+import seedu.address.model.deck.entry.Entry;
+import seedu.address.model.deck.entry.UniqueEntryList;
+
+/**
+ * Represents a deck of flashcards
+ */
+public class Deck {
+
+    private final DeckName deckName;
+    private UniqueEntryList cards;
+
+    /**
+     * Name must be present and not null
+     * @param deckName name of the deck of cards
+     */
+    public Deck(DeckName deckName) {
+        requireNonNull(deckName);
+        this.deckName = deckName;
+        this.cards = new UniqueEntryList();
+    }
+
+    public DeckName getDeckName() {
+        return this.deckName;
+    }
+
+    public UniqueEntryList getCards() {
+        return this.cards;
+    }
+
+    /**
+     * Two decks are considered to be the same if they have the same name
+     * This defines a notion of equality between the two decks
+     *
+     * @param otherDeck to be compared with the current deck
+     * @return true if both decks have the same name
+     */
+    public boolean isSameDeck(Deck otherDeck) {
+        if (otherDeck == this) {
+            return true;
+        }
+        return otherDeck != null
+                && otherDeck.getDeckName().equals(getDeckName());
+    }
+
+    /**
+     * Returns true if both decks have the same name
+     * This defines a notion of equality between two deck objects
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Deck)) {
+            return false;
+        }
+
+        Deck otherDeck = (Deck) other;
+        return otherDeck.getDeckName().equals(getDeckName());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(deckName, cards);
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getDeckName())
+                .append(" Cards: ");
```

Comment:
Hmm indentation looks a bit off - not sure if it's just GitHub.
Did you `CMD/CTRL+ALT+L` in Intellij Idea to autoformat?

Reply:
The indentation doesn't change, seems to be that way. In other classes like Entry.java as well

Reply:
gotcha